### PR TITLE
Plain (No-Auth) War file

### PR DESCRIPTION
**Pull request by matzew**
* Refactoring to include a plain (no-auth) war file
* applya flat role model (just users), if desired, on the new WAR
* Keycloak and roles are still supported
* Version bump to 2.0.0
* Docker DEV Image for the "plain" image

**MISSING:** Docker dev image for the traditional KC war...

### CODE REVIEW

File: service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/SearchManager.java [hunk 1]
```diff
@@ -70,9 +75,22 @@ public PushSearchService getSearchService() {
     @Produces
     @LoggedIn
     public String extractUsername() {
-        KeycloakPrincipal p = (KeycloakPrincipal) httpServletRequest.getUserPrincipal();
-        KeycloakSecurityContext kcSecurityContext = p.getKeycloakSecurityContext();
-        return kcSecurityContext.getToken().getPreferredUsername();
 
+        final KeycloakPrincipal principal = (KeycloakPrincipal) httpServletRequest.getUserPrincipal();
```
[the remark this answers is not in the document]
Comment (matzew) [reply service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/SearchManager.java]: we do have that code already in the system - it's nothing new

[the remark this answers is not in the document]
Comment (matzew) [reply service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/SearchManager.java]: I could see if an SPI works here.   

in each WAR we have a different "SearchManager"

[the remark this answers is not in the document]
Comment (matzew) [reply service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/SearchManager.java]: sounds to me more an enhancement JIRA, of the existing code

**Comment by wtrocki** [comment]
Looks great. Verification steps before we merge that?

**Comment by matzew** [comment]
@wtrocki the `no-auth` plain case is:

The quickest is:
```
mvn clean install -DskipTests && docker run -p 18081:8080 -it aerogear/ups:plain 
```

than go to `http://localhost:18081/` and create an Application.

in the log you should see some:

```
...
14:11:52,506 DEBUG [org.jboss.aerogear.unifiedpush.service.impl.SearchManager] (default task-1) Running outside of Keycloak context
...
```

NOTE: I am now working on a container for UPS-KC :)

**Comment by pb82** [comment]
@matzew Verified locally. I can see the in the logs:

```
08:24:42,626 DEBUG [org.jboss.aerogear.unifiedpush.service.impl.SearchManager] (default task-2) Running outside of Keycloak context
08:24:42,627 DEBUG [org.jboss.aerogear.unifiedpush.service.impl.SearchManager] (default task-2) Running without any Auth context
```

Creating an application and variant worked. The only thing is that the log is full of those messages, but that should be fine in production when the `DEBUG` level is not visible.

**Pull request merged**